### PR TITLE
BC-2603 - task create

### PR DESCRIPTION
**Pull request by odalys-dataport**
# Short Description

## Links to Ticket and related Pull-Requests
BC-2603
https://github.com/hpi-schul-cloud/schulcloud-server/pull/3694

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/utils/store-accessor.ts [hunk 1]
```diff
@@ -37,6 +38,7 @@ let roomModule: RoomModule;
 let roomsModule: RoomsModule;
 let schoolsModule: SchoolsModule;
 let statusAlertsModule: StatusAlertsModule;
+let tasksModule: TasksModule;
```
Comment (virgilchiriac) [inline src/utils/store-accessor.ts:41]: shoudl this be singluar, is for Task?

**Comment by MartinSchuhmacher** [comment]
I still am unsure about this whole task/tasks separation to be honest, even though we do the same for room/rooms, which does not mean we should continue with doing so. I find it very confusing actually (which is the same with room/rooms), so maybe we can talk about that.

File: src/pages/TaskCreate.page.vue [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+		const breadcrumbs = [
+			{
+				text: i18n.t("common.words.tasks"),
+				to: "/tasks",
+			},
+		];
+
+		const save = () => {
+			taskModule.createTask({
```
Comment (christian-darsow) [inline src/pages/TaskCreate.page.vue:42]: I'm also getting confused now :) While I see good reasons to have two stores for single/multiple I'm wondering if this more generic action should not be in the tasks store (like the deleteTask action)

Comment (MartinSchuhmacher) [reply src/pages/TaskCreate.page.vue:42]: agree to that

File: src/store/task.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+		if (!this._taskApi) {
+			this._taskApi = TaskApiFactory(undefined, "/v3", $axios);
+		}
+		return this._taskApi;
+	}
+
+	@Action
+	async createTask(params: TaskCreateParams): Promise<void> {
+		// check for permission here or in page?
+		// probably checked in server?
```
Comment (virgilchiriac) [inline src/store/task.ts:42]: server authorization is done in any case, the logic should therefor be more on the level of page load / show button

**Review by MartinSchuhmacher — APPROVED** [review]

**Pull request merged**